app/generators/radio_generator.rb

```diff
@@ -49,14 +49,19 @@ def decorate_with_stop_subscription(question, option_body, option)
   end
 
   def question_options(question, option, elem_id)
-    {
-      name: answer_name(idify(question[:id])),
-      type: 'radio',
-      id: elem_id,
-      value: option[:raw][:value].presence || option[:raw][:title],
-      required: !(question.key?(:required) && question[:required] == false),
-      class: 'validate'
-    }
+    value = option[:raw][:value].presence || option[:raw][:title]
+    decorate_with_previous_value(
+      question,
+      question[:id],
+      {
+        name: answer_name(idify(question[:id])),
+        type: 'radio',
+        id: elem_id,
+        value: value,
+        required: !(question.key?(:required) && question[:required] == false),
+        class: 'validate'
+      }
+    ) { |previous_value| [:checked, previous_value == value] }
```

Comment:
A good test to add (if you don't have it already) might be to have a radio question with a prefilled option selected that has a `shows_questions` attribute, and then checking if the questions in that `shows_questions` thing actually show up when the questionnaire is loaded (I think they should, but it is something we'd want to test).